VAULT-46794: add SCIM changes to 2.0.4 docs (#2832)
hashicorp/web-unified-docs: content/vault/v2.x/content/api-docs/secret/identity/scim.mdx

- `userName` maps to the entity name.
- `externalId` maps to the entity external ID.
- `active` maps to the inverse of the entity `disabled` state.
- `groups` is a read-only field that lists direct memberships in groups that

>   the same SCIM client manages (memberships to unmanaged groups are not listed).

Reviewer: ```suggestion
  the same SCIM client manages (the list excludes memberships in unmanaged groups).
```
Style correction: write in active voice

- `displayName` maps to the group name.
- `members[].value` must contain Vault user entity IDs.

> Each SCIM client can work only with members it owns. You

Reviewer: ```suggestion
Each SCIM client only works with members it owns. You
```
Style correction: use simple sentence construction

- Vault does not support sort, ETag, or change password operations.
- User `PATCH` only supports a single `replace` operation for `active`.
- Group `PATCH` only supports member add and remove operations.
- Vault does not support arbitrary SCIM filter expressions beyond the
  documented fields and operators for `/Users` and `/Groups`.
- Group members must use Vault user entity IDs, not `userName` or
  `externalId`.
- User `groups` only shows direct memberships in groups that the same SCIM

>   client manages.

Reviewer: ```suggestion
- User `groups` only shows direct memberships in groups managed by the same SCIM
  client.
```
The sentence as written was a little hard to parse. Let me know if the new sentence inadvertently changes the meaning.
